File: src/main/java/com/Khata/Khata/Controller/GigController.java
package com.Khata.Khata.Controller;

import com.Khata.Khata.Entity.Gig;
import com.Khata.Khata.Service.GigService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/Gig")
public class GigController
{
    private GigService gigService;

    GigController(GigService gigService)
    {
        this.gigService = gigService;
    }

    @PostMapping("/add")
    public ResponseEntity<Gig> addGig(@RequestBody Gig gig)
    {
        Gig savedGig = gigService.addGig(gig);
        return ResponseEntity.ok(savedGig);
    }
}

File: classtracker/src/main/kotlin/com/classtrackeer/classtracker/Controller/GigController.kt
package com.classtrackeer.classtracker.Controller

import com.classtrackeer.classtracker.Model.Gig
import com.classtrackeer.classtracker.Service.GigService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/gig")
class GigController (@Autowired val gigService : GigService){

    @PostMapping("/addGig")
    fun addGig(@RequestBody gig : Gig) : ResponseEntity<Gig>{
        val savedGig : Gig = gigService.addGig(gig)
        return ResponseEntity(savedGig, HttpStatus.OK)
    }
}